packages/ckeditor5-dev-env/lib/release-tools/tasks/update-package-versions.js

```diff
@@ -0,0 +1,257 @@
+				} );
+			} else {
+				fs.writeFileSync( file, newFileData, 'utf-8' );
+			}
+		}
+	}
+
+	return { found: packageJsonArray.length, updated: updatedFiles, differences };
+}
+
+/**
+ * Updates the CKEditor 5 dependencies, except the `*-dev` and `*-inspector` ones, in the provided `package.json` file.
+ *
+ * @param {Object} parsedPkgJson Object to update.
+ * @param {String} propertyName Name of the property to update.
+ */
+function updateObjectProperty( parsedPkgJson, propertyName ) {
+	const version = parsedPkgJson.version;
+	const regex = /^@ckeditor\/ckeditor5-([a-z]+)|^ckeditor5$/;
+	const exceptions = [
+		'dev',
+		'inspector',
+		'react',
+		'vue',
+		'angular'
+	];
+
+	for ( const dependency in parsedPkgJson[ propertyName ] ) {
+		const match = dependency.match( regex );
+
+		if ( !match || exceptions.includes( match[ 1 ] ) ) {
+			continue;
+		}
+
+		parsedPkgJson[ propertyName ][ dependency ] = `^${ version }`;
+	}
+}
+
+/**
+ * Takes in raw changelog for a single file generated by `diff` library, and returns formatted array of strings, containing
+ * human-readable, line by line changelog of a file, with removals and additions colored. If the file has any longer parts without changes,
+ * these will be partially hidden.
+ *
+ * @param {Array<Object>} diff Array of changes for a single file generated by `diff` library.
+ * @returns {Array<String>} Formatted changelog split into single lines.
+ */
+function formatDiff( diff ) {
+	const formattedDiff = [];
+	const regex = /(?<=":) (?=")/;
+
+	for ( let i = 0; i < diff.length; i++ ) {
+		const previous = diff[ i - 1 ];
+		const current = diff[ i ];
+		const next = diff[ i + 1 ];
+		const currentLines = current.value.split( '\n' );
+
+		const validReplaceSequence = current.removed && next.added && regex.test( current.value ) && regex.test( next.value );
+
+		if ( validReplaceSequence ) {
+			// Adding removals followed by additions (formatted).
+			formattedDiff.push( [
+				current.value.split( regex )[ 0 ],
+				' ',
+				chalk.red( current.value.split( regex )[ 1 ] ),
+				chalk.green( next.value.split( regex )[ 1 ] )
+			].join( '' ) );
+
+			i++;
+		} else if ( !current.added && !current.removed && currentLines.length > 8 ) {
+			// Cutting out the middle of a long streak of unchanged lines.
+			const shortenedLines = [
+				...currentLines.slice( 0, 3 ),
+				chalk.gray( `[...${ currentLines.length - 7 } lines without changes...]` ),
+				...currentLines.slice( -4 )
+			].join( '\n' );
+
+			formattedDiff.push( shortenedLines );
+		} else if ( current.added ) {
+			// Other additions (trimming whitespaces in replacements)
+			formattedDiff.push( chalk.green( previous && previous.removed ? current.value.trim() : current.value ) );
+		} else if ( current.removed ) {
+			// Other removals
+			formattedDiff.push( chalk.red( current.value ) );
+		} else {
+			// Unchanged lines
+			formattedDiff.push( current.value );
+		}
```

Comment:
There is quite a lot of conditions. Could it be simplified?

For example, probably the order of conditions could be better:

```
if ( validReplaceSequence ) {
    // ...
} else if ( current.added ) {
    // ...
} else if ( current.removed ) {
    // ...
} else if ( currentLines.length > 8 ) {
    // ...
} else {
    // ...
}
```